syntax = "proto3";

option objc_class_prefix = "FS";

package fileservice;

service FileService {
  rpc ListFiles (ListFilesRequest) returns (FilesReply) {}
  rpc SearchFiles (SearchFilesRequest) returns (FilesReply) {}
}

message ListFilesRequest {
  optional int32 limit = 1;
}

message SearchFilesRequest {
  string pattern = 1;
  optional int32 limit = 2;
}

message FilesReply {
  string file_list = 1;
}
